### PR TITLE
Enable opaque-keys

### PR DESCRIPTION
**Pull request by valarauca**
Ensure compatibility with `opaque-keys` PR.

Changes don't break library even if with `opaque-key` disabled

https://github.com/slog-rs/slog/pull/153

### CODE REVIEW

File: lib.rs [hunk 1]
```diff
@@ -70,7 +71,12 @@ impl<S: serde::Serializer> SerdeSerializer<S> {
 
 macro_rules! impl_m(
     ($s:expr, $key:expr, $val:expr) => ({
-        try!($s.ser_map.serialize_entry($key, $val)
+        #[cfg(feature = "opaque-keys")]
```
Comment (dpc) [inline lib.rs]: If we implement `as_ref() -> &str` https://doc.rust-lang.org/std/primitive.str.html#impl-AsRef%3Cstr%3E for opaque `Key`, we wouldn't have to have conditional compilation here and `key.as_ref()` would work in both versions.

Comment (valarauca) [reply lib.rs]: Good catch I didn't think of that.

**Comment by dpc** [comment]
Landed to master. I'm sorry that I didn't cherry-pick, but it was just easier. Thanks!

**Comment by valarauca** [comment]
No problem. Just glad it made it in. 

Sorry for the radio silence on the main issue. I got wrapped up in a larger work project rewriting one of micro-services. I might get a chance to revise the main PR over the holidays.

**Pull request closed without merging**